src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -146,6 +152,7 @@ private static void initHttpClient(String accountName) {
     HttpClientBuilder clientBuilder =
         HttpClientBuilder.create()
             .setConnectionManager(connectionManager)
+            .evictIdleConnections(DEFAULT_EVICT_IDLE_AFTER_SECONDS, TimeUnit.SECONDS)
```

Comment:
In retry logic, we should consume the response to not let any connection block other http requests. I will open the PR. but this will also help alleviate the problem.